chia/simulator/full_node_simulator.py

```diff
@@ -1,26 +1,35 @@
-from typing import List, Optional
+from typing import Dict, List, Optional
 
 from chia.consensus.block_record import BlockRecord
 from chia.consensus.multiprocess_validation import PreValidationResult
+from chia.full_node.full_node import FullNode
 from chia.full_node.full_node_api import FullNodeAPI
 from chia.protocols.full_node_protocol import RespondBlock
 from chia.simulator.simulator_protocol import FarmNewBlockProtocol, ReorgProtocol
+from chia.types.blockchain_format.sized_bytes import bytes32
 from chia.types.full_block import FullBlock
 from chia.util.api_decorators import api_request
+from chia.util.config import lock_and_load_config, save_config
 from chia.util.ints import uint8
+from tests.block_tools import BlockTools
 
 
 class FullNodeSimulator(FullNodeAPI):
-    def __init__(self, full_node, block_tools) -> None:
+    def __init__(self, full_node: FullNode, block_tools: BlockTools, config: Dict) -> None:
         super().__init__(full_node)
         self.bt = block_tools
         self.full_node = full_node
-        self.config = full_node.config
+        self.config = config
         self.time_per_block = None
-        if "simulation" in self.config and self.config["simulation"] is True:
+        self.full_node.simulator_transaction_callback = self.autofarm_transaction
+        if "simulation" in self.config["full_node"] and self.config["full_node"]["simulation"] is True:
             self.use_current_time = True
         else:
             self.use_current_time = False
+        if "auto_farm" in self.config["simulator"] and self.config["simulator"]["auto_farm"] is True:
```

Comment:
you can do: 
```
auto_farm = self.config["simulator"].get("auto_farm", False)
```

Reply:
that's a default value in case it's not present. Same for config["simulator"]

Reply:
Yeah, i forgot about that. Thanks!